# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/ffvpx/libavcodec/threadframe.h
# Commit: 186e2a334f2d
# Full Hash: 186e2a334f2deb37b29f44b28bc8c83babeeb9a4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-09 09:37:17
# Regressor Bug: 1877221
# File Overlap Count: 1
# Description:
#   Bug 1877221 - Pull in new ffmpeg code, reapply patch. r=media-playback-reviewers,chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199909
# ==============================================================================

diff -r c5920de78d16 -r 186e2a334f2d media/ffvpx/libavcodec/threadframe.h
--- a/media/ffvpx/libavcodec/threadframe.h	Thu Feb 08 10:39:53 2024 +0000
+++ b/media/ffvpx/libavcodec/threadframe.h	Thu Feb 08 10:39:54 2024 +0000
@@ -27,9 +27,7 @@
 typedef struct ThreadFrame {
     AVFrame *f;
     AVCodecContext *owner[2];
-    // progress->data is an array of 2 ints holding progress for top/bottom
-    // fields
-    AVBufferRef *progress;
+    struct ThreadFrameProgress *progress;
 } ThreadFrame;
 
 /**
@@ -80,13 +78,10 @@
  * @param avctx The current context.
  * @param f The picture being released.
  */
-void ff_thread_release_ext_buffer(AVCodecContext *avctx, ThreadFrame *f);
+void ff_thread_release_ext_buffer(ThreadFrame *f);
 
 int ff_thread_ref_frame(ThreadFrame *dst, const ThreadFrame *src);
 
-int ff_thread_replace_frame(AVCodecContext *avctx, ThreadFrame *dst,
-                            const ThreadFrame *src);
-
-int ff_thread_can_start_frame(AVCodecContext *avctx);
+int ff_thread_replace_frame(ThreadFrame *dst, const ThreadFrame *src);
 
 #endif